### ISSUE TITLE
Implement Types support in JSDoc for Swagger gen

### ISSUE THREAD

**Issue by vrudikov** [thiagobustamante/typescript-rest-swagger#9]
Hi, Thiago
So i tried to return declared Type from API method and added type's reference to JSDoc, but swagger generator can't handle it. It just writes 'Inline Model 1':
<img width="360" alt="image" src="https://user-images.githubusercontent.com/493714/28943814-d220a8e6-78a8-11e7-86ea-d0a488f46714.png">

You can see examples [here](https://github.com/vrudikov/typescript-rest-boilerplate/blob/master/src/controllers/hello-object-controller.ts#L29) and [here](https://github.com/vrudikov/typescript-rest-boilerplate/blob/master/src/controllers/hello-object-controller.ts#L44)

**Comment by vrudikov** [comment]
@thiagobustamante Hi. Any updates? If you are busy - please suggest me a string or filename. I will do a research

**Comment by thiagobustamante** [comment]
Hi, @vrudikov ,

he problema is not related to JsDocs. The problem is that typescript-rest-swagger can not understand this declaration:

```typescript
export type SimpleHelloType = {
    greeting: string;
};
```

if you use interfaces or classes, it works:

**Comment by vrudikov** [comment]
@thiagobustamante A-W-E-S-O-M-E! Thank you

> The problem is not related to JsDocs. The problem is that typescript-rest-swagger can not understand this declaration
Yeah. I knew that. But wanted to understand where it should be fixed

**Comment by vrudikov** [comment]
Just a second. I think i found a bug. I will try to created a PR and will post a fix

**Comment by vrudikov** [comment]
@thiagobustamante Hello!
Could you try to replace your object with this one:
```
export type SimpleHelloType = {
    /**
     * Description for greeting property
     */
    greeting: string;
    arrayOfSomething: Something[];

    profile: {
        name: string,
        gender: string,
        location: string,
        website: string,
        picture: string
    };

    comparePassword: (candidatePassword: string, cb: (err: any, isMatch: any) => {}) => void
};

export type Something = {
    someone: string,
    kind: string
};
```

And fire swagger generator again, plz

**Comment by thiagobustamante** [comment]
Yes you are right. I added it to the test case and I am reopening the issue.

**Comment by krazar** [comment]
Is it the same reason why it does not work on types in interface ?
```typescript
export type GroupId = number;

export interface Group {
  id: GroupId;
  groupName: string;
}
```

Also, In my case, this object is not directly linked to the rest layer.  Is typescript-rest-swagger look for every defined type in the project ?

**Comment by vrudikov** [comment]
@thiagobustamante hi. have you had a time to look at my case? do you need any help?

**Comment by ngraef** [comment]
@vrudikov This should be fixed by recent commits. Please try with the latest version (`0.0.12`) and let me know if you still have issues.

@krazar Your issue is a bit different. I'm looking into it.

**Comment by vrudikov** [comment]
@ngraef Awesome. Thank you. Will take a look today

**Comment by vrudikov** [comment]
Wow. Forgot to look at it. Will do in near future
